src/main/java/io/github/antishake/CircularBuffer.java

```diff
@@ -11,43 +10,48 @@
         -will maintain write pointer when updating with the latest values
  */
 public class CircularBuffer {
-    public int read_pointer,write_pointer=0;
-    public int size_buffer =201;
-    double[] circularbuffer =new double[size_buffer];
-    boolean buffer_full_once=false;
-    public boolean isEmpty(){
-        return read_pointer==write_pointer;
+  private int readPointer, writePointer = 0;
+  private int sizeBuffer = 201;
+  private Coordinate[] circularbuffer = new Coordinate[sizeBuffer];
+  private boolean bufferFullOnce = false;
+
+  public boolean isEmpty() {
+    return readPointer == writePointer;
+  }
+
+  public void add(Coordinate element) {
+    if (readPointer < sizeBuffer) {
+      if (!bufferFullOnce) {
+        circularbuffer[readPointer] = element;
+        readPointer++;
+      }
+      if (bufferFullOnce) {
+        circularbuffer[readPointer] = element;
+        readPointer++;
+        writePointer++;
```

Comment:
Here, I can see the value is written based on read pointer and write pointer is updated only when the buffer is full.
Dont think, write pointer will have the correct value. Please check.